src/main/java/org/jenkinsci/account/Application.java

```diff
@@ -168,6 +169,8 @@ public HttpResponse doDoSignup(
             throw new UserError("Last name is required");
         if (isEmpty(email))
             throw new UserError("e-mail is required");
+        if (!email.equals(emailconfirm))
+            throw new UserError("e-mail is not matching");
```

Comment:
Perhaps *the two email addresses are not matching*? Error messages can be too short :-)